### DISCUSSION TITLE
group-focus-within

### DISCUSSION THREAD

**Discussion by yainspan** [tailwindlabs/tailwindcss#4591] [Ideas]
I ran across this when building a form. It would be nice to have something that combines `group-focus` and `focus-within`. This seems to be relatively simple, and could work something like this:
```
<label class="group">
    <span class="group-focus-within:text-red-500">Label Text</span>
    <input type="text">
</label>
```

Then, when the input gains focus, the `span` would be red from the focus within the `.group`.

The way `focus-within` works means that it must be applied to a parent element, so we can't target child elements when a sibling gains focus.

**Comment by VicGUTT** [comment]
Hey !

I'm not entirely sure but what you're looking for seems to be the "**peer**" selector coming to Tailwind v2.2 in a couple days. You can take a look at the merged pull request [here](https://github.com/tailwindlabs/tailwindcss/pull/4556).

Anyway here's a [quick plugin](https://play.tailwindcss.com/Yu5C1of15J?file=config) for you that does just that.

Hope it helps :v:

**Reply by yainspan** [reply]
Thanks!
I'm not sure if the peer selector would fit in exactly in my use case, but the plugin works perfectly! I forgot how customizable things are 😄 .

**Reply by VicGUTT** [reply]
Ahah my pleasure :)

**Reply by VicGUTT** [reply]
It'd be cool if you could mark the discussion as "answered" :)

**Comment by adamwathan** [comment]
Hey! If you use the new [JIT engine](https://tailwindcss.com/docs/just-in-time-mode) this actually already works, along with `group-active`, `group-disabled`, `group-required`, and every other pseudo-class 👍🏻

**Reply by VicGUTT** [reply]
That's cool. I did take a look at the [JIT: Add exhaustive pseudo-class and pseudo-element variant support](https://github.com/tailwindlabs/tailwindcss/pull/4482/files) PR before answering but didn't see `focus-within` in the list. Looking at the code though, I do see it was already supported !

Very cool and thanks for chiming in :v:

**Comment by mrnasil** [comment]
2 Solution


https://play.tailwindcss.com/Wa9I0jJNW0
